File: app/src/main/java/com/huwei/sweetmusicplayer/business/fragments/LocalArtistFragment.java
package com.huwei.sweetmusicplayer.business.fragments;

import android.support.v4.app.FragmentManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.StaggeredGridLayoutManager;
import android.view.View;

import com.huwei.sweetmusicplayer.business.LocalMusicActivity;
import com.huwei.sweetmusicplayer.R;
import com.huwei.sweetmusicplayer.contains.IMusicViewTypeContain;
import com.huwei.sweetmusicplayer.business.fragments.base.BaseFragment;
import com.huwei.sweetmusicplayer.business.ui.adapters.LocArtistInfoAdapter;
import com.huwei.sweetmusicplayer.business.ui.adapters.RecyclerViewAdapterBase;
import com.huwei.sweetmusicplayer.util.MusicUtils;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.Bean;
import org.androidannotations.annotations.EFragment;
import org.androidannotations.annotations.ViewById;

/**
 * 本地音乐的Artist列表页面
 */
@EFragment(R.layout.fragment_artists)
public class LocalArtistFragment extends BaseFragment implements IMusicViewTypeContain{

	@ViewById
	RecyclerView rv_artist;

	@Bean
    LocArtistInfoAdapter adapter;

	FragmentManager fragmentManager;

	@AfterViews
	void init(){
		fragmentManager=getActivity().getSupportFragmentManager();

		adapter.setData(MusicUtils.queryArtistList());
        rv_artist.setLayoutManager(new StaggeredGridLayoutManager(2,StaggeredGridLayoutManager.VERTICAL));
        rv_artist.setAdapter(adapter);
		adapter.setOnItemClickListener(new RecyclerViewAdapterBase.OnItemClickListener() {
			@Override
			public void onItemClick(View view, int position) {
				startActivity(LocalMusicActivity.Companion.getStartActIntent(mAct, SHOW_MUSIC_BY_ARTIST,
						adapter.getData().get(position).getArtist(), adapter.getData().get(position).getArtistId()));
			}
		});
	}
}

File: app/src/main/java/com/huwei/sweetmusicplayer/business/fragments/LocalArtistFragment.kt
package com.huwei.sweetmusicplayer.business.fragments

import android.os.Bundle
import android.support.v4.app.FragmentManager
import android.support.v7.widget.StaggeredGridLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.huwei.sweetmusicplayer.business.LocalMusicActivity
import com.huwei.sweetmusicplayer.R
import com.huwei.sweetmusicplayer.contains.IMusicViewTypeContain
import com.huwei.sweetmusicplayer.business.fragments.base.BaseFragment
import com.huwei.sweetmusicplayer.business.ui.adapters.LocArtistInfoAdapter
import com.huwei.sweetmusicplayer.util.MusicUtils
import kotlinx.android.synthetic.main.fragment_artists.*

/**
 * 本地音乐的Artist列表页面
 */
class LocalArtistFragment : BaseFragment(), IMusicViewTypeContain {
    internal var adapter: LocArtistInfoAdapter? = null

    lateinit internal var fragmentManager: FragmentManager

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return LayoutInflater.from(context).inflate(R.layout.fragment_artists, null)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        fragmentManager = activity.supportFragmentManager

        adapter = LocArtistInfoAdapter(context)
        adapter!!.setData(MusicUtils.queryArtistList())
        rv_artist!!.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)
        rv_artist!!.adapter = adapter
        adapter!!.setOnItemClickListener { view, position ->
            startActivity(LocalMusicActivity.getStartActIntent(mAct, IMusicViewTypeContain.SHOW_MUSIC_BY_ARTIST,
                    adapter!!.data[position].artist, adapter!!.data[position].artistId!!))
        }
    }
}
